### PR TITLE
Rework chk_decision to suck less

### PR DESCRIPTION
**Pull request by skizzerz**
Not ready to merge just yet, still WIP

### CODE REVIEW

File: src/status/absent.py [hunk 1]
```diff
@@ -21,14 +20,13 @@ def add_absent(var, target, reason):
                 del var.VOTES[votee]
             break
 
+def get_absent(var):
+	return set(ABSENT.keys())
```
Comment (Vgr255) [inline src/status/absent.py]: No need for `.keys()` call here.

File: src/status/influence.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+__all__ = ["add_influence", "remove_influence", "get_influence"]
+
+INFLUENCE = UserDict() # type: UserDict[users.User, int]
+
+def add_influence(var, target, amount=1):
+    if target not in get_players():
+        return
+
+    INFLUENCE[target] = INFLUENCE.get(target, 1) + amount
+	if INFLUENCE[target] == 1:
```
Comment (Vgr255) [inline src/status/influence.py]: Did you mess up indentation or is GitHub being weird?

File: src/status/votes.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+def force_abstain(var, votee):
+	evt = Event("force_vote", {})
+	if not evt.dispatch(var, votee, None):
+		return
+
+	FORCED_COUNTS[votee] = FORCED_COUNTS.get(votee, 0) - 1
+	if FORCED_COUNTS[votee] == 0:
+		del FORCED_COUNTS[votee]
+
```
Comment (Vgr255) [inline src/status/votes.py]: Merge `force_vote` and `force_abstain` into a single `add_weight` (or `_add_weight`) function, then have both `force_vote` and `force_abstain` call it with proper parameters.

File: src/votes.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+    """Vote for a game mode if no game is running, or for a player to be lynched."""
+    if message:
+        if var.PHASE == "join" and wrapper.public:
+            from src.wolfgame import game
+            return game.caller(wrapper.client, wrapper.source.nick, wrapper.target.name, message)
+        return lynch.caller(wrapper.client, wrapper.source.nick, wrapper.target.name, message)
+    return show_votes.caller(wrapper.client, wrapper.source.nick, wrapper.target.name, message)
+
+# Specify force=True to force a lynch even if there is no majority
+def chk_decision(var, *, force=False):
```
Comment (skizzerz) [inline src/votes.py]: Keep the kwarg as timeout and not force. They're different.

File: src/votes.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+            channels.Main.send(messages["village_abstain"])
+
+            from src.wolfgame import transition_night
+            transition_night()
+
+        if to_vote:
+            global LYNCHED
+            LYNCHED += len(to_vote) # track how many people we've lynched today
+
+            if force:
```
Comment (skizzerz) [inline src/votes.py]: This should be "if timeout"

File: src/votes.py [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+                        an = "n" if rrole.startswith(("a", "e", "i", "o", "u")) else ""
+                        lmsg = random.choice(messages["lynch_reveal"]).format(votee, an, rrole)
+                    else:
+                        lmsg = random.choice(messages["lynch_no_reveal"]).format(votee)
+                    channels.Main.send(lmsg)
+                    add_dying(var, votee, "villager", "lynch")
+
+            kill_players(var, end_game=False) # FIXME
+
+        elif force:
```
Comment (skizzerz) [inline src/votes.py]: "elif timeout"

**Pull request merged**